### PR TITLE
parsers: add do and declare statement

### PR DESCRIPTION
**Pull request by TszKitLo40**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5184 

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 1]
```diff
@@ -6854,7 +6878,30 @@ char_type:
...
+do_stmt:
+    DO expression_list
+    {
+        $$ = &tree.Do {
+            Exprs: $2,
+        }
+    }
 
+declare_stmt:
+    DECLARE variable_list
```
Comment (iamlinjunhong) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:6890]: use var_name_list:
var_name_list:
      var_name
      {...}
|.     var_name ',' var_name
      {....}

Comment (iamlinjunhong) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:6890]: lack of type
type:
    Any valid MySQL data type
you can use column_type

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 2]
```diff
@@ -495,7 +503,7 @@ import (
 %type <str> field_terminator starting_opt lines_terminated_opt
 %type <lines> load_lines export_lines_opt
 %type <int64Val> ignore_lines
-%type <varExpr> user_variable variable system_variable
+%type <varExpr> user_variable variable system_variable local_variable
```
Comment (iamlinjunhong) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:506]: remove local_variable

Comment (iamlinjunhong) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:506]: declare:
%type <strs> var_name_list

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 3]
```diff
@@ -1547,6 +1558,17 @@ var_name:
         $$ = $1 + "." + $3
     }
 
+var_name_list:
+    var_name
+    {
+        $$ = []tree.Identifier{tree.Identifier($1)}
```
Comment (iamlinjunhong) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:1564]: $$ = []string{$1}

**Review by iamlinjunhong — APPROVED** [review]

File: test/cases/util/declare.result [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (ouyuanning) [inline test/cases/util/declare.result:1]: why empty?

File: test/cases/util/declare.sql [hunk 5]
```diff
@@ -0,0 +1,2 @@
+DECLARE a, b INT
```
Comment (ouyuanning) [inline test/cases/util/declare.sql:1]: should end with ";”？

File: test/cases/util/declare.result (hunk 4 above)
Comment (TszKitLo40) [reply test/cases/util/declare.result:1]: fixed

File: test/cases/util/declare.sql (hunk 5 above)
Comment (TszKitLo40) [reply test/cases/util/declare.sql]: fixed

**Review by ouyuanning — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

File: test/cases/util/declare.result [hunk 6]
```diff
@@ -0,0 +1,4 @@
+DECLARE a, b INT;
+not supported: declare a b int default null
```
Comment (fengttt) [inline test/cases/util/declare.result:2]: I am confused by these PRs, so the test still says not supported, I am not sure this is what we want.

Comment (TszKitLo40) [reply test/cases/util/declare.result:2]: This is a matrix camp task, and we just need to implement the parser.

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]
OK, please leave a commit message and in the issue, stating this is parsing only and we have not done the execution part yet.

Thank you.

**Review by daviszhen — APPROVED** [review]

**Pull request merged**